### PR TITLE
Small improvements for kevm

### PR DESCRIPTION
**Pull request by anvacaru**
- closes: #1472 
 - removes two unused addresses: the `Hardhat console` and the `Foundry caller` which are currently not used.
 - Refactor `Vm2.sol` interface into a `KEVMCheats` contract which includes the cheat codes from the old interface and a new helper function `notBuiltinAddress`.

### CODE REVIEW

**Comment by ehildenb** [comment]
Let's add the Solidity hook `isBuiltinAddress` as well on this PR.

File: include/kframework/serialization.md [hunk 1]
```diff
@@ -53,8 +53,8 @@ Address/Hash Helpers
     rule #sender("")  => .Account
     rule #sender(STR) => #addr(#parseHexWord(Keccak256(STR))) requires STR =/=String ""
 
-    syntax Int ::= #addrFromPrivateKey ( String ) [function]
- // --------------------------------------------------------
+    syntax Int ::= #addrFromPrivateKey ( String ) [function, concrete]
+ // ------------------------------------------------------------------
     rule #addrFromPrivateKey ( KEY ) => #addr( #parseHexWord( Keccak256 ( Hex2Raw( ECDSAPubKey( Hex2Raw( KEY ) ) ) ) ) )
 ```
```
Comment (ehildenb) [inline include/kframework/serialization.md:59]: This isn't the right way to mark it as concrete I don't think. Instead, name the rule `rule [addrFromPrivateKey]: ...`, and add the rule name to `tests/specs/concrete-rules.txt`, and to `KEVM.concrete_rules()` in `kevm.py`.

**Comment by ehildenb** [comment]
@anvacaru see the last two commits that I had to add.

**Review by ehildenb — APPROVED** [review]

**Comment by ehildenb** [comment]
This appears to be a massive performance improvement!

File: include/kframework/serialization.md [hunk 2]
```diff
@@ -53,9 +53,9 @@ Address/Hash Helpers
     rule #sender("")  => .Account
     rule #sender(STR) => #addr(#parseHexWord(Keccak256(STR))) requires STR =/=String ""
 
-    syntax Int ::= #addrFromPrivateKey ( String ) [function]
- // --------------------------------------------------------
-    rule #addrFromPrivateKey ( KEY ) => #addr( #parseHexWord( Keccak256 ( Hex2Raw( ECDSAPubKey( Hex2Raw( KEY ) ) ) ) ) )
+    syntax Int ::= #addrFromPrivateKey ( String ) [function, klabel(addrFromPrivateKey)]
+ // ------------------------------------------------------------------------------------
+    rule [addrFromPrivateKey]: #addrFromPrivateKey ( KEY ) => #addr( #parseHexWord( Keccak256 ( Hex2Raw( ECDSAPubKey( Hex2Raw( KEY ) ) ) ) ) )
```
Comment (anvacaru) [inline include/kframework/serialization.md:58]: @ehildenb is there any difference in using the rule label and using the `klabel` tag? Also, is the klabel required anymore?

Comment (ehildenb) [reply include/kframework/serialization.md:58]: klabel is not required, and yes there is a big difference. One is a label for referring to the production, and the other is a rule label for referring to the rule.

Comment (ehildenb) [reply include/kframework/serialization.md:58]: No need to remove the klabel.

Comment (ehildenb) [reply include/kframework/serialization.md:58]: You can find documentation on both in the K user manual.

**Pull request merged**